documentation/src/main/asciidoc/topics/con_write_behind.adoc

```diff
@@ -1,15 +1,14 @@
 [id='write_behind']
 = Write-Behind
-Write-Behind is an cache writing mode where writes to memory are synchronous
-and writes to cache stores are asynchronous. When a client application
-updates a cache entry, {brandname} adds the update to a modification queue and
-then modifies the cache store in a different thread than the client thread.
-
-You can configure the number of threads that consume the modification queue and
-apply updates to the underlying cache store. The modification queue fills up if
-there are not enough threads to handle the updates or if the underlying cache
-store becomes unavailable. When this occurs, {brandname} uses Write-Through
-mode until the modification queue can accept new entries.
+Write-Behind is a cache writing mode where writes to memory are synchronous
+and writes to cache stores are asynchronous. When client applications
+update entries, {brandname} modifies cache stores in different threads to the
+client threads.
+
+When clients send write requests, {brandname} adds those operations to a
+modification queue until each previous write operation completes. If the 
+modification queue becomes full, subsequent write operations block the calling
```

Comment:
We will never block the calling thread, that is the new advantage of the SPI is it allows you to do to this and must. The way write behind works is there is a queue of modifications (which can be empty) and there may be a pending batch of operations being currently applied. The way it works is a write operation is added to the queue and if there is no pending batch it is submitted to be processed. The caller is never blocked during this and the operation completes immediately. If the pending batch is still running, the queue may fill up in the interim, when it does fill additional write operations are also enqueued but we do not allow the operation to complete immediately. Instead it will complete when the pending batch completes. We do this in a non blocking way, so the invoking thread is never blocked though.

Reply:
@wburns argh. I thought I left a question on this one because I thought the same about the non-blocking SPI and probably got mixed up in a chat. I guess I didn't properly save the question in the review. I blame github. anyway, thx for the detailed explanation. hopefully it's better now.